### DISCUSSION TITLE
How does rq work under the hood?

### DISCUSSION THREAD

**Discussion by AnastasiusVivaldus** [rq/rq#1589] [General]
I'd like to understand please how rq works. I have a docker-compose set up with a flask API that will enqueue jobs, and then a worker in a separate container that executes the function defined within the flask-API code. I assume that the only way the worker container and flask container communicate is via redis. But that must mean that the code for the worker function is passed via redis. Is that correct? If so, in what format is it passed? As a string? Thanks for clarification on how this works under the hood.

**Comment by rpkak** [comment]
The worker has to be executed in a source directory. In redis is written, what function and which args to use, but the source code must be on the workers site.
